src/quota.rs

```diff
@@ -124,14 +124,16 @@ impl Context {
     /// in case for some providers the quota is always at ~100%
     /// and new space is allocated as needed.
     pub(crate) async fn update_recent_quota(&self, session: &mut ImapSession) -> Result<()> {
+        info!(self, "Updating IMAP quota.");
+
         let quota = if session.can_check_quota() {
             let folders = get_watched_folders(self).await?;
-            get_unique_quota_roots_and_usage(session, folders).await
+            Some(get_unique_quota_roots_and_usage(session, folders).await?)
```

Comment:
So, does this mean that if a provider erreneously says that it has the quota capability, while it actually does not, connecting to this provider will fail?

Not sure whether this ever happens, but also, not sure whether we should include this right before the release - otoh, it hopefully also fixes a bug.

Reply:
Before, `quota.modified` was updated in this case, maybe still do so and then bubble up the error? This way we won't try to update the quota again after reconnecting. This would be closer to the previous behavior